client/src/components/ListCard.js

```diff
@@ -56,6 +61,17 @@ const ListCard = (props) => {
     setAddProductOpen((previous) => !previous);
   };
 
+  const getItemCount = async () => {
+    const res = await fetch(`/list-to-products/${listId}`)
+    const json = await res.json();
+    setItemCount(json.length)
+  };
+
+  useEffect(() => {
+    getItemCount()
+  }, [listOpen,listChange]);
+
```

Comment:
You are making too many API calls. We most likely want all of the relevant data fetched once, when the page loads, at the top level component that needs any of that data (dashboard?). So here for example, we should have access to the list details and when the list changes we can just set itemCount to the length of the new list that's chosen rather than make another API call.